src/Resource.php

```diff
@@ -80,8 +80,16 @@ public function toArray()
             $array['relationships'] = $relationships;
         }
 
+        $links = [];
         if (! empty($this->links)) {
-            $array['links'] = $this->links;
+            $links = $this->links;
+        }
+        $serializerLinks = $this->serializer->getLinks($this->data);
+        if (! empty($serializerLinks)) {
+            $links = array_merge($serializerLinks, $links);
+        }
```

Comment:
That's what I assumed (and what is covered by a test), but frankly I wasn't really sure about it.. And after thinking about it again it might be more flexible if a serializer can overrule resource links.
In the original patch by @bnchdrff it was also the other way around - should I swap it?